File: frontend/licenta/src/app/services/review.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import baseUrl from "./helper";

@Injectable({
  providedIn: 'root'
})
export class ReviewService {
  constructor(private http: HttpClient) { }

  public addReview(review: any) {
    return this.http.post(`${baseUrl}/review/`, review);
  }

  public updateReview(review: any) {
    return this.http.put(`${baseUrl}/review/`, review);
  }

  public getAllReviews() {
    return this.http.get(`${baseUrl}/review/`);
  }

  public getReviewById(id: number) {
    return this.http.get(`${baseUrl}/review/${id}`);
  }

  public getReviewsByUniversityId(universityId: number) {
    return this.http.get(`${baseUrl}/review/university/${universityId}`);
  }

  public getReviewsByAuthorId(authorId: number) {
    return this.http.get(`${baseUrl}/review/author/${authorId}`);
  }

  public getReviewsByUniversityIdAndAuthorId(universityId: number, authorId: number) {
    return this.http.get(`${baseUrl}/review/university/${universityId}/author/${authorId}`);
  }

  public deleteReviewById(id: number) {
    return this.http.delete(`${baseUrl}/review/${id}`);
  }

  public likeReview(reviewId: number, userId: number) {
    return this.http.put(`${baseUrl}/review/${reviewId}/like/${userId}`, null);
  }

  public getReviewsLikedByUser(userId: number) {
    return this.http.get(`${baseUrl}/review/liked-reviews/${userId}`);
  }
}

File: src/main/java/licenta/service/ReviewService.java
package licenta.service;

import licenta.model.Review;
import org.springframework.stereotype.Service;

import java.util.List;

public interface ReviewService {
    Review createReview(Review review);

    Review updateReview(Review review) throws Exception;

    Review getReview(Long id) throws Exception;

    List<Review> getReviewsByUniversityId(Long universityId) throws Exception;

    List<Review> getReviewsByAuthorId(Long authorId) throws Exception;

    List<Review> getReviewsByUniversityIdAndAuthorId(Long universityId, Long authorId) throws Exception;

    List<Review> getReviews();

    void deleteReview(Long id);

    Review likeReview(Long reviewId, Long userId) throws Exception;

    Review dislikeReview(Long reviewId, Long userId) throws Exception;

    boolean getLikeStatus(Long reviewId, Long userId) throws Exception;

    List<Review> getLikedReviews(Long userId) throws Exception;
}
